### ISSUE TITLE
Windows 10, Nightcode Crashing

### ISSUE THREAD

**Issue by mudgen** [oakes/Nightcode#230]
I am a new user of Nightcode.  I installed the latest version of Nightcode 2.4.0 on Windows 10.  I imported an existing boot project and that worked. When I start a repl it starts up and soon after that Nightcode disappears, is no longer running, appears to crash.

How do I debug this?

Also, how do I start a new boot project within Nightcode?

**Comment by oakes** [comment]
Does it crash when you run any other command in the build pane, or just the repl?

**Comment by mudgen** [comment]
It crashes when I try to build or start the repl.

**Comment by oakes** [comment]
Do you know what JDK version is installed? And is the boot project you're using available on github anywhere? I'm trying to reproduce but have not been able to yet.

**Comment by mudgen** [comment]
Yes. Java version 1.8.0_141.

The boot project is not on github.   

Here is the build.build file contents:
(def project 'ignition-project)
(def version "0.1.0-SNAPSHOT")

(set-env! :resource-paths #{"resources" "src"}
          :source-paths   #{"test"}
          :dependencies   '[[org.clojure/clojure "1.8.0"]
                            [clj-http "3.7.0"]
                            [nightlight "1.9.1" :scope "test"]])

(task-options!
 pom {:project     project
      :version     version
      :description "FIXME: write description"
      :url         "http://example/FIXME"
      :scm         {:url "https://github.com/yourname/ignition-project"}
      :license     {"Eclipse Public License"
                    "http://www.eclipse.org/legal/epl-v10.html"}})

(deftask build
  "Build and install the project locally."
  []
  (comp (pom) (jar) (install)))

Here is the ignition-project/core.clj file contents:
(ns ignition-project.core
  (:require [boot.core :as boot :refer [deftask get-env]]))

(defn foo
  "I don't do a whole lot."
  [x]
  (println x "Hello, World!"))

(defn reload [] (load-file "build.boot"))

(def m "test")

**Comment by mudgen** [comment]
Here is a link to a gist on github that shows the contents of the files with indentation:
https://gist.github.com/mudgen/65b6cd67b83220f9c1587313e1fe3efe

**Comment by mudgen** [comment]
If you want we can do a Teamviewer meeting and I can show you my screen so you can see what is happening on my computer.

**Comment by oakes** [comment]
Could you please try running the JAR version of Nightcode from cmd.exe, and see if it crashes as well? I wonder if we would at least see useful output in the console after the crash. I tried building a project with the exact same `build.boot` contents you posted without any issue, so that is the nest debugging idea I have in mind.

**Comment by mudgen** [comment]
Good idea. I executed Nightcode-2.4.0.jar on the command line.  Nothing is printed there but the application is no longer crashing. I can start up the boot REPL.  I can type in something at the REPL like `(println "something")` but when I press enter nothing happens. I also noticed that when the boot REPL starts up a background java process starts on my machine that constantly consuming 18 percent of the CPU.

**Comment by oakes** [comment]
Ah yeah the boot REPL appears to be broken on Windows, I'll look into that. So it looks like the crashing of the installed version is a separate issue. When you installed 2.4.0, did you have an older version installed previously? I noticed that there were problems with installing 2.4.0 on top of an older version, because I changed the bundled runtime to Java 9. If you haven't done so already, try completely uninstalling Nightcode and then installing it again.
